### PR TITLE
make (no) space after type assertion user configurable

### LINKED ISSUE #10362: VSCode "Format Code" causes tslint failure with (old-school) type assertions
Preceded by https://github.com/Microsoft/vscode-tslint/issues/69 and https://github.com/Microsoft/vscode/issues/10296

**TypeScript Version:**  1.8.0

**Code**

If I have code like this:

```ts
const bar = <Bar> Thing.getFoo();
```

and then "Format Code" in VSCode (alt-shift-F on Mac), I get this:

```ts
const bar = <Bar>Thing.getFoo();
```

**Expected behavior:**

The spacing should not have been changed. (__Note__: without a [specific setting](http://palantir.github.io/tslint/rules/whitespace/), this causes a failure with tslint.)

**Actual behavior:** 

The space after the old-school/bracket-style type assertion is removed, causing a tslint error.

### PR DESCRIPTION
**Pull request by gnalck**
Fixes #10362

### CODE REVIEW

File: src/services/services.ts [hunk 1]
```diff
@@ -1356,6 +1356,7 @@ namespace ts {
         InsertSpaceAfterOpeningAndBeforeClosingNonemptyBrackets: boolean;
         InsertSpaceAfterOpeningAndBeforeClosingTemplateStringBraces: boolean;
         InsertSpaceAfterOpeningAndBeforeClosingJsxExpressionBraces?: boolean;
+        InsertSpaceAfterTypeAssertion: boolean;
```
Comment (mhegazy) [inline src/services/services.ts]: this should be optional, otherwise all users of the API have to specify it.

**Comment by mhegazy** [comment]
@gnalck thanks for the PR, and extremely sorry for the delay.

can you refresh the PR against master. modulo one comment, I think this PR is ready to go.

**Comment by mhegazy** [comment]
manually merged. thanks!

**Comment by gnalck** [comment]
sweet, thanks!

**Comment by ydogandjiev** [comment]
@mhegazy, @gnalck, hey guys, I'm super excited to use this feature in order to make the auto-format functionality compliant with my team's tslint rules. Unfortunately, I can't seem to set the insertSpaceAfterTypeAssertion option in my VSCode settings. It doesn't appear as an option and even if I manually enter it, the setting seems to have no effect:
    "typescript.format.insertSpaceAfterTypeAssertion": true,

What am I missing here? Has this feature been officially released yet?

**Comment by csentis** [comment]
@ydogandjiev I had asked the same in Microsoft/vscode#10296 - glad to see that it got assigned to @mjbvz. Looking forward to this implemented, too!

**Comment by lucap86** [comment]
I have a trouble with this feature.

with
```
let a = <number[]>["1", "2", "3", "4"]
```
no space added after <number[]>

otherwise if i have
```
let a = <number>"1"
```
it correctly add a space after <number>

**Pull request merged**